execution_policy: Regression for OpenMP and TBB backends

Recently, this commit [here](https://github.com/thrust/thrust/commit/1669350bcbc026e2df10ab75bbc4f088761024d1) introduced a regression in the `execution_policy` module. In particular, I now get the following compilation error with the OpenMP (and TBB) backend:

```
thrust/thrust/execution_policy.h:347:41: error: the type ‘const device_t {aka const thrust::system::omp::detail::par_t}’ of constexpr variable ‘thrust::device’ is not literal
 THRUST_INLINE_CONSTANT detail::device_t device;
                                         ^~~~~~
In file included from thrust/thrust/system/omp/execution_policy.h:29:0,
                 from thrust/thrust/execution_policy.h:37,
thrust/thrust/system/omp/detail/par.h:33:8: note: ‘thrust::system::omp::detail::par_t’ is not literal because:
 struct par_t : thrust::system::omp::detail::execution_policy<par_t>,
        ^~~~~
thrust/thrust/system/omp/detail/par.h:33:8: note:   ‘thrust::system::omp::detail::par_t’ is not an aggregate, does not have a trivial default constructor, and has no constexpr constructor that is not a copy or move constructor
```

--- a/thrust/system/omp/detail/par.h
+++ b/thrust/system/omp/detail/par.h
@@ -35,7 +35,7 @@ struct par_t : thrust::system::omp::detail::execution_policy<par_t>,
     thrust::system::omp::detail::execution_policy>
 {
   __host__ __device__
-  par_t() : thrust::system::omp::detail::execution_policy<par_t>() {}
+  THRUST_CONSTEXPR par_t() : thrust::system::omp::detail::execution_policy<par_t>() {}
 };
 
 

--- a/thrust/system/tbb/detail/par.h
+++ b/thrust/system/tbb/detail/par.h
@@ -35,7 +35,7 @@ struct par_t : thrust::system::tbb::detail::execution_policy<par_t>,
     thrust::system::tbb::detail::execution_policy>
 {
   __host__ __device__
-  par_t() : thrust::system::tbb::detail::execution_policy<par_t>() {}
+  THRUST_CONSTEXPR par_t() : thrust::system::tbb::detail::execution_policy<par_t>() {}
 };
 
 

